### PR TITLE
refactor: change where we stick millw

### PR DESCRIPTION
**Pull request by ckipp01**
Instead of creating a temp dir for millw for Mill users this instead
uses a more stable location of just sticking it in `.metals/millw`. This
_should_ help with some changes coming in Mill and millw spawning from https://github.com/com-lihaoyi/mill/discussions/2079.

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/builds/MillBuildTool.scala [hunk 1]
```diff
@@ -42,7 +43,8 @@ case class MillBuildTool(userConfig: () => UserConfiguration)
     val millWrapper =
       if (Properties.isWin) "millw.bat"
       else "millw"
-    val out = BuildTool.copyFromResource(tempDir, millWrapper)
+    val out =
+      BuildTool.copyFromResource(RelativePath(".metals").toNIO, millWrapper)
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/builds/MillBuildTool.scala:47]: Shouldn't we use:
```suggestion
      BuildTool.copyFromResource(workspace.resolve(".metals").toNIO, millWrapper)
```

Relative path should probably start from `.`, but that might not always be the case.

**Review by tgodzik — CHANGES_REQUESTED** [review]
Thanks for working on this!

File: metals/src/main/scala/scala/meta/internal/builds/MillBuildTool.scala (hunk 1 above)
Comment (ckipp01) [reply metals/src/main/scala/scala/meta/internal/builds/MillBuildTool.scala]: Ah yea, good call. Switched it.

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**